app/models/async_transaction/base.rb

```diff
@@ -13,6 +15,8 @@ class Base < ApplicationRecord
     }
 
     attr_encrypted :metadata, key: Settings.db_encryption_key
+    serialize :metadata, JsonMarshal::Marshaller
```

Comment:
AsyncTransaction::Base record metadata is an array (verified in prod and staging. The metadata needs to be coerced into a string before saving into the db and retrieved as an array. The JsonMarshal::Marshaller class will properly save and retrieve in the expected form.